src/main/java/de/tum/in/www1/artemis/repository/ModelElementRepository.java

```diff
@@ -20,4 +23,46 @@
     ModelElement findByModelElementIdWithCluster(@Param("modelElementId") String elementId);
 
     List<ModelElement> findByModelElementIdIn(List<String> elementIds);
+
+    /**
+     * Interface used to define return type for `countOtherElementsInClusterBySubmissionId`
+     */
+    interface ModelElementCount {
+
+        String getElementId();
+
+        Long getNumberOfOtherElements();
+    }
+
+    /**
+     * For the given Submission `id` returns a list of raw object array representing two columns.
+     * First index/column corresponds to the TextBlock `id` while the second one corresponds to
+     * the number of other blocks in the same cluster as given block with id = `id`.
+     * For all TextBlock's of the Submission with the given `id`
+     * finds their respective cluster and retrieves the number of other blocks in the same cluster
+     * @param submissionId the id of the Submission
+     * @return the number of other TextBlock's in the same cluster as the block with given `id`
```

Comment:
```suggestion
     * First index/column corresponds to the TextBlock `id` while the second one corresponds to
     * the number of other blocks in the same cluster as given block with id = `id`.
     * For all TextBlock's of the Submission with the given `id`
     * finds their respective cluster and retrieves the number of other blocks in the same cluster
     * @param submissionId the id of the Submission
     * @return the number of other TextBlock's in the same cluster as the block with given `id`
```

TextBlock -> Model ?